flink-table-store-core/src/main/java/org/apache/flink/table/store/file/data/AppendOnlyWriter.java

```diff
@@ -88,6 +81,9 @@ public void write(RowData rowData) throws Exception {
                 "Append-only writer can only accept insert row kind, but current row kind is: %s",
                 rowData.getRowKind());
         writer.write(rowData);
+        if (!toCompact.isEmpty()) {
+            submitCompaction();
```

Comment:
submit compaction per record?
I think should be in rolling file or prepareCommit.

Reply:
Actually, the criteria for compaction are actively evaluated (as you say, per record to check `toCompact` and submit if not empty), but the true compaction rewrite frequency is unchanged. I'm both ok.